packages/compass-components/src/components/actions/types.ts

```diff
@@ -0,0 +1,37 @@
+import type { glyphs } from '@leafygreen-ui/icon';
+
+import type { ItemActionButtonSize } from './constants';
+
+export type ItemBase<Action extends string> = {
```

Comment:
nit: I don't think as a rule we usually have the types separated into their own file.

Reply:
The main reason for this file is because two mutually independent files (`item-action-menu.tsx` and `item-action-group.tsx`) have declarations which union this type. I don't want one to be considered more important by the other importing from it and I don't want to hoist it as that'll introduce a circular import.